src/identity/oauth.ts

```diff
@@ -175,10 +181,26 @@ export class OAuthProvider implements Provider {
   }
 
   // Open the supplied URL in the user's default browser
-  // TODO: probably not cross-platform
   private async openURL(url: string) {
     return new Promise<void>((resolve, reject) => {
-      child_process.exec(`open "${url}"`, undefined, (err) => {
+      let open = null;
+      let command = `"${url}"`;
+      switch (process.platform) {
+        case 'darwin':
+          open = 'open';
+          break;
+        case 'linux' || 'freebsd' || 'netbsd' || 'openbsd':
+          open = 'xdg-open';
+          break;
+        case 'win32':
+          open = 'start';
+          command = `"" ${command}`;
+          break;
+        default:
+          return reject(`OAuth: unsupported platform: ${process.platform}`);
+      }
+      console.log(`Your browser will now be opened to: ${url}`);
```

Comment:
I think we might want to send this to stderr w/ `console.error` (similar to how it's done in [cosign](https://github.com/sigstore/sigstore/blob/ca3b544f985f87114485ffa686dd558f80357449/pkg/oauth/oidc/interactive.go#L186)). That way if you're redirecting the output of the command to a file to capture the signature/attestation you won't also get this string.

Reply:
Great catch!